### PR TITLE
ET-4334_ca_on_ajax_fix

### PR DESCRIPTION
**Pull request by akarshikalowe**
fixed the scraper

### CODE REVIEW

**Comment by akarshikalowe** [comment]
fix to ca_on_ajax

File: ca_on_ajax/people.py [hunk 1]
```diff
@@ -1,49 +1,43 @@
...
         page = self.lxmlize(COUNCIL_PAGE)
 
-        councillors = page.xpath('//ul[@class="subNav top"]/li/ul//li/a[contains(text(), "Councillor")] | //ul[@class="subNav top"]/li/ul//li/a[contains(text(), "Mayor")]')
-        assert len(councillors), 'No councillors found'
+        councillors = page.xpath('//table[@class="councilTable"]')
         for councillor in councillors:
-            name = councillor.text_content()
+            image = councillor.xpath('.//@src')[0]
+            councillor_name = councillor.xpath('.//tr/td[1]/p[1]/img/@alt')
+            assert len(councillors), 'No councillors found'
```
Comment (jpmckinney) [inline ca_on_ajax/people.py]: This should be outside the for loop.

File: ca_on_ajax/people.py [hunk 2]
```diff
@@ -1,49 +1,43 @@
...
 
-            url = councillor.attrib['href']
-            page = self.lxmlize(url)
+            for i in councillor_name:
```
Comment (jpmckinney) [inline ca_on_ajax/people.py]: Why is this a for loop? Shouldn't we just add `[0]` at the end of the `councillor_name = …` line above?

File: ca_on_ajax/people.py [hunk 3]
```diff
@@ -1,49 +1,43 @@
...
+                name = i.split('-', 1)[0].strip()
+                district = i.split('Councillor ')[-1].strip()
 
-            if councillor == councillors[0]:
+            if name == councillor_name[0]:
```
Comment (jpmckinney) [inline ca_on_ajax/people.py]: A better test would be `if 'Mayor' in councillor_name[0]` (or, if we add `[0]` to the assignment as described in another comment, just `if 'Mayor' in councillor_name`).

File: ca_on_ajax/people.py [hunk 4]
```diff
@@ -1,49 +1,43 @@
...
                 district = 'Ajax'
                 role = 'Mayor'
+
             else:
-                district = re.findall(r'Ward.*', page.xpath('//div[@id="printAreaContent"]//h1')[0].text_content())[0].strip()
-                role = page.xpath('//div[@id="printAreaContent"]//h1')[0].text_content()
-                role = re.search('((?:Regional )?Councillor)', role).group(1)
+                role = 'Councillor'
```
Comment (jpmckinney) [inline ca_on_ajax/people.py]: This doesn't account for the fact that there are regional councillors.

File: ca_on_ajax/people.py [hunk 5]
```diff
@@ -1,49 +1,43 @@
...
-                else:
-                    contact = line.xpath('./td[2]/a/@href')[0]
-                    p.add_link(contact)
+            p.image = image
+
+            if phone:
+                p.add_contact('cell', phone, 'legislature')
+            if tel:
+
+                p.add_contact('email', self.get_email(councillor))
```
Comment (jpmckinney) [inline ca_on_ajax/people.py]: I don't understand the logic… if you file a telephone number, you add an email? What happens to the telephone number? Also, please remove the newline after the if-condition.

**Pull request merged**